src/scicookie/{{cookiecutter.project_slug}}/build-system/hatchling-pyproject.toml

```diff
@@ -0,0 +1,135 @@
+[build-system]
+requires = ["hatchling>=1.17.1", "hatch-vcs>=0.3.0"]
```

Comment:
Hmm, if we are keeping `VCS` versioning as the default one, it might make sense to add a `version.pyi` file. But as I look at other backends, in particular `setuptools`, there is no usage of `setuptools_scm`, which means we might not want the `VCS` versioning by default?

Reply:
This would require adding something like this -

```toml
[tool.hatch]
version.source = "vcs"
build.hooks.vcs.version-file = "some/path/here/version.py"
```

Reply:
@Saransh-cpp I removed the `vcs` version as the default one. Is it necessary to add this snippet? (as `vcs` is not in use)
```
[tool.hatch]
version.source = "vcs"
build.hooks.vcs.version-file = "some/path/here/version.py"
```

Reply:
Not now! But we might need to explore the path to provide a user with an option to turn on VCS versioning. cc: @xmnlab.